backend/YouTubeV2.Application/Services/VideoServices/VideoService.cs

```diff
@@ -48,5 +48,27 @@ public async Task SetVideoProcessingProgressAsync(Video video, ProcessingProgres
             video.ProcessingProgress = processingProgress;
             await _context.SaveChangesAsync(cancellationToken);
         }
+
+        public async Task<VideoMetadataDto> GetVideoMetadataAsync(Guid id, CancellationToken cancellationToken = default)
```

Comment:
probably we are missing the logic of not allowing other users to get info about private videos. I believe it is also missing in getting video stream so you can fix it